packages/consumption/src/modules/requests/itemProcessors/createAttribute/CreateAttributeRequestItemProcessor.ts

```diff
@@ -45,23 +49,67 @@ export class CreateAttributeRequestItemProcessor extends GenericRequestItemProce
             );
         }
 
-        if (recipientIsAttributeOwner || senderIsAttributeOwner || ownerIsEmptyString) {
-            return ValidationResult.success();
-        }
+        if (!(recipientIsAttributeOwner || senderIsAttributeOwner || ownerIsEmptyString)) {
+            if (typeof recipient !== "undefined") {
+                return ValidationResult.error(
+                    ConsumptionCoreErrors.requests.invalidRequestItem(
+                        "The owner of the provided RelationshipAttribute for the `attribute` property can only be the Sender's Address, the Recipient's Address or an empty string. The latter will default to the Recipient's Address."
+                    )
+                );
+            }
 
-        if (typeof recipient !== "undefined") {
             return ValidationResult.error(
                 ConsumptionCoreErrors.requests.invalidRequestItem(
-                    "The owner of the provided RelationshipAttribute for the `attribute` property can only be the Sender's Address, the Recipient's Address or an empty string. The latter will default to the Recipient's Address."
+                    "The owner of the provided RelationshipAttribute for the `attribute` property can only be the Sender's Address or an empty string. The latter will default to the Recipient's Address."
                 )
             );
         }
 
-        return ValidationResult.error(
-            ConsumptionCoreErrors.requests.invalidRequestItem(
-                "The owner of the provided RelationshipAttribute for the `attribute` property can only be the Sender's Address or an empty string. The latter will default to the Recipient's Address."
-            )
-        );
```

Comment:
I see. Now this comments on the current state, but your changes make it obvious in the diff: Why do we throw different, but almost identical, error messages depending on whether recipient is defined or not? In my opinion it would be much easier to not check for the recipient and just throw the first error message. If the recipient is not defined and, hence, their address not known, the reader will know what to do from reading the error message. This would save us complexity in the code. Also while we're at it, we could write "address" lowercase. Maybe these changes could be done in a small separate PR. What do you think?

Reply:
I also think that this is currently cumbersome and we should definitely do it differently :D I think we should take care of this in another PR, probably in the long run we should prevent that the address of the Recipient can be set explicitly to ensure consistency with the behavior of other RequestItems. Also I'd like your suggestion to write address lowercase.